android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -0,0 +1,400 @@
+  protected void finishSharedAnimation(int tag) {
+    View view = mCurrentSharedTransitionViews.get(tag);
+    if (view != null) {
+      ((ViewGroup) mTransitionContainer).removeView(view);
+      View parentView = mSharedTransitionParent.get(view.getId());
+      int childIndex = mSharedTransitionInParentIndex.get(view.getId());
+      ViewGroup parentViewGroup = ((ViewGroup) parentView);
+      if (childIndex <= parentViewGroup.getChildCount()) {
+        parentViewGroup.addView(view, childIndex);
+      } else {
+        parentViewGroup.addView(view);
+      }
+      Snapshot viewSourcePreviousSnapshot = mSnapshotRegistry.get(view.getId());
+      int originY = viewSourcePreviousSnapshot.originY;
+      if (findStack(view) == null) {
+        viewSourcePreviousSnapshot.originY = viewSourcePreviousSnapshot.topInsetFromParent;
+      }
+      Map<String, Object> snapshotMap = viewSourcePreviousSnapshot.toBasicMap();
+      Map<String, Object> preparedValues = new HashMap<>();
+      for (String key : snapshotMap.keySet()) {
+        Object value = snapshotMap.get(key);
+        preparedValues.put(key, (double) PixelUtil.toDIPFromPixel((int) value));
+      }
+      mAnimationsManager.progressLayoutAnimation(view.getId(), preparedValues, true);
+      viewSourcePreviousSnapshot.originY = originY;
+      mCurrentSharedTransitionViews.remove(view.getId());
+    }
+    if (mCurrentSharedTransitionViews.isEmpty()) {
+      mSharedTransitionParent.clear();
+      mSharedTransitionInParentIndex.clear();
+      if (mTransitionContainer != null) {
+        ViewParent transitionContainerParent = mTransitionContainer.getParent();
+        if (transitionContainerParent != null) {
+          ((ViewGroup) transitionContainerParent).removeView(mTransitionContainer);
+        }
+      }
+      mIsSharedTransitionActive = false;
+    }
+  }
+
+  @Nullable
+  private View findScreen(View view) {
+    ViewParent parent = view.getParent();
+    while (parent != null) {
+      if (parent.getClass().getSimpleName().equals("Screen")) {
+        return (View) parent;
+      }
+      parent = parent.getParent();
+    }
+    return null;
+  }
+
+  @Nullable
+  private View findStack(View view) {
+    ViewParent parent = view.getParent();
+    while (parent != null) {
+      if (parent.getClass().getSimpleName().equals("ScreenStack")) {
+        return (View) parent;
+      }
+      parent = parent.getParent();
+    }
+    return null;
+  }
+
+  protected void makeSnapshot(View view) {
+    Snapshot snapshot = new Snapshot(view);
+    View screen = findScreen(view);
+    if (screen != null) {
+      snapshot.originY -= screen.getTop();
+    }
+    mSnapshotRegistry.put(view.getId(), snapshot);
+  }
+
+  protected void visitTreeForTags(int[] viewTags, boolean clearConfig) {
+    if (viewTags == null) {
+      return;
+    }
+    for (int viewTag : viewTags) {
+      visitTree(viewTag, clearConfig);
+    }
+  }
+
+  private void visitTree(int tag, boolean clearConfig) {
```

Comment:
I would refactor this whole function a bit.

Like I mentioned in another comment, I don't think there's any need to take the `tag` here, we can pass the `View` instead and get the tag from it.

Another thing is, the `clearConfig` param here is only checked in one `if` statement, and passed down in recursion - it would be more readable to have two functions, one for `clearConfig` case and one for `!clearConfig` case instead.
Then, you'd get a bit of boilerplate for getting `viewGroupManager` and recursing into children common between those two functions, so what you can do instead is have the one `visitTree` function which takes the `View` to start from, and a callback to call on each of its descendants in the tree instead - again I think it would be more readable, than a single function doing two different things depending on a boolean flag.